legal-api/src/legal_api/resources/v2/business/business_filings/business_documents.py

```diff
@@ -70,23 +71,32 @@ def get_documents(identifier: str, filing_id: int, legal_filing_name: str = None
                                       **{'filing_id': filing_id, 'identifier': identifier})
         ), HTTPStatus.NOT_FOUND
 
-    if not legal_filing_name:
+    if not legal_filing_name and not file_key:
         if identifier.startswith('T') and filing.status == Filing.Status.COMPLETED:
-            return {}, HTTPStatus.NOT_FOUND
+            return {'documents': {}}, HTTPStatus.OK
         return _get_document_list(business, filing)
 
-    if legal_filing_name and ('application/pdf' in request.accept_mimetypes):
-        if legal_filing_name.lower().startswith('receipt'):
-            return _get_receipt(business, filing, jwt.get_token_auth_header())
-
-        return get_pdf(filing.storage, legal_filing_name)
+    if 'application/pdf' in request.accept_mimetypes:
+        if legal_filing_name:
+            if legal_filing_name.lower().startswith('receipt'):
+                return _get_receipt(business, filing, jwt.get_token_auth_header())
+
+            return get_pdf(filing.storage, legal_filing_name)
+        elif file_key and (document := Document.find_by_file_key(file_key)):
+            if document.filing_id == filing.id:  # make sure the file belongs to this filing
+                response = MinioService.get_file(document.file_key)
```

Comment:
How long is the download link good for?

Reply:
this function returns the file not the link (so no expiry)